components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+        String exportedFilePath;
+        Application applicationDetails = null;
+        File exportedApplicationArchiveFile = null;
+        String exportedAppDirName = "exported-application";
+        String pathToExportDir = System.getProperty(RestApiConstants.JAVA_IO_TMPDIR) + File.separator +
+                "exported-app-archives-" +
+                UUID.randomUUID().toString(); //creates a directory in default temporary-file directory
+        String username = RestApiUtil.getLoggedInUsername();
+        String exportedFileName = null;
+        if (StringUtils.isBlank(appName) || StringUtils.isBlank(appOwner)) {
+            RestApiUtil.handleBadRequest("Application name or owner should not be empty or null.", log);
+        }
+        try {
+            consumer = RestApiUtil.getConsumer(username);
+            FileBasedApplicationImportExportManager importExportManager = new FileBasedApplicationImportExportManager
+                    (consumer, pathToExportDir);
+            if (importExportManager.isOwnerAvailable(appOwner)) {
+                applicationDetails = importExportManager.getApplicationDetails(appName, appOwner);
+            }
+            if (applicationDetails == null) {
+                // 404
+                String errorMsg = "No application found with name " + appName + " owned by " + appOwner;
+                log.error(errorMsg);
+                return Response.status(Response.Status.NOT_FOUND).entity(errorMsg).build();
+            } else if (!MultitenantUtils.getTenantDomain(applicationDetails.getSubscriber().getName()).equals
+                    (MultitenantUtils.getTenantDomain(username))) {
+                String errorMsg = "Cross Tenant Exports are not allowed";
+                log.error(errorMsg);
+                return Response.status(Response.Status.FORBIDDEN).entity(errorMsg).build();
+            }
+            exportedFilePath = importExportManager.exportApplication(applicationDetails, exportedAppDirName);
+            String zippedFilePath = importExportManager.createArchiveFromExportedAppArtifacts(exportedFilePath,
+                    pathToExportDir, exportedAppDirName);
+            exportedApplicationArchiveFile = new File(zippedFilePath);
+            exportedFileName = exportedApplicationArchiveFile.getName();
+        } catch (APIManagementException e) {
+            RestApiUtil
+                    .handleInternalServerError("Error while exporting Application: " + appName, e, log);
+        }
+        Response.ResponseBuilder responseBuilder = Response.status(Response.Status.OK)
+                .entity(exportedApplicationArchiveFile).type(MediaType.APPLICATION_OCTET_STREAM);
+        responseBuilder
+                .header("Content-Disposition", "attachment; filename=\"" + exportedFileName + "\"");
+        return responseBuilder.build();
+    }
+
+    @Override
+    public String exportApplicationsGetGetLastUpdatedTime(String appName, String appOwner) {
+        return null;
```

Comment:
why this method return null?

Reply:
This generated method cannot be implemented